### PR TITLE
initial commit for median value calculation via R-ArcGIS bridge blog pos

### PR DESCRIPTION
**Pull request by merit-amystuyv**
initial commit for median value calculation via R-ArcGIS bridge blog post - please review and let me know if I can clarify anything or make any changes

### CODE REVIEW

File: content/blog/summarize_within_median_r_arcgis/index.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
+---
+date: "2020-01-29T10:50:00Z"
```
Comment (merit-mthompson) [inline content/blog/summarize_within_median_r_arcgis/index.md:2]: Think we want the year to be 2021

```suggestion
date: "2021-01-29T10:50:00Z"
```

**Comment by merit-mthompson** [comment]
Not covered in the directions but for the images we're embedding in this repo we want to resize them a bit so they're not so large.

Here's the same image shrunken from ~5MB to ~256K; can we update/replace?

nathan-dumlao-qq9XjVZe4Lk-unsplash

**Comment by merit-amystuyv** [comment]
no problem, just updated the image file

**Review by merit-mthompson — CHANGES_REQUESTED** [review]
Looks awesome Amy, nice work! My only suggestion is the date on the post, otherwise it looks perfect.

**Review by merit-mthompson — APPROVED** [review]

**Comment by merit-mthompson** [comment]
Looks good to me! Adding others from PSWD as reviewers for the final go-ahead.

**Review by merit-rluk — APPROVED** [review]
Looks great, Amy! I'm going to comment on a few typos but overall I'd say it's good to go.

File: content/blog/summarize_within_median_r_arcgis/index.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
+---
+date: "2021-01-29T10:50:00Z"
+description: "Learn how to summarize the median value of your point features within a polygon feature class in ArcGIS Pro using the R-ArcGIS bridge."
+featuredImage: nathan-dumlao-qq9XjVZe4Lk-unsplash.jpg
+featuredImageCredit: "Nathan Dumlao"
+featuredImageLink: "https://unsplash.com/photos/qq9XjVZe4Lk"
+tags: ["gis", "R", "Esri", "ArcGIS Pro", "statistics"]
+title: Summarize Median Values in ArcGIS Pro via R-ArcGIS Bridge
+---
+_If you've ever longed for 'median' as an option in the dropdown list of Esri's 'Summarize Within' tool when working with non-normally distributed data, long no more! Use this script and the R-ArcGIS Bridge to fulfill your non-parimetric function needs._
```
Comment (merit-rluk) [inline content/blog/summarize_within_median_r_arcgis/index.md:10]:
```suggestion
_If you've ever longed for 'median' as an option in the dropdown list of Esri's 'Summarize Within' tool when working with non-normally distributed data, long no more! Use this script and the R-ArcGIS Bridge to fulfill your non-parametric function needs._
```

File: content/blog/summarize_within_median_r_arcgis/index.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+There is no question that this data does not satisfy the requirements of normally distributed data, meaning we should ***not*** use a parametric test (e.g. `mean`) to analyze the data. Using the `mean` function when aggregating these values up to the census block level would atrificially push your average towards those outliers, misrepresenting what the 'average' person in that census block is really experiencing. When the purpose of this analysis is to provide data-driven insight into where new broadband infrastructure investments should be made, pushing the metric towards those high outliers is absolutely unacceptable. Fortunately, the median value of your data provides one option to better represent the true middle ground of what users on the ground are experiencing.
```
Comment (merit-rluk) [inline content/blog/summarize_within_median_r_arcgis/index.md:20]:
```suggestion
There is no question that this data does not satisfy the requirements of normally distributed data, meaning we should ***not*** use a parametric test (e.g. `mean`) to analyze the data. Using the `mean` function when aggregating these values up to the census block level would artificially push your average towards those outliers, misrepresenting what the 'average' person in that census block is really experiencing. When the purpose of this analysis is to provide data-driven insight into where new broadband infrastructure investments should be made, pushing the metric towards those high outliers is absolutely unacceptable. Fortunately, the median value of your data provides one option to better represent the true middle ground of what users on the ground are experiencing. 
```

File: content/blog/summarize_within_median_r_arcgis/index.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+This particular project has many components and is largely working in ArcGIS Pro's model builder with some deeper statistics and figure generation happening in RStudio. Our goal is to have cohesion between these two programs, so we did not want a workflow that required us to run some analysis in ArcGIS Pro, leave this program and open up RStudio to run a custom-made script, and then go back into ArcGIS Pro to run the rest of the model. Luckily, Esri's R-ArcGIS Bridge allows us to create the custom tool you'll see below and then import that into the ArcGIS Pro interface whenever we want to calculate the median download and uplaod speeds for each census block in our study area.
```
Comment (merit-rluk) [inline content/blog/summarize_within_median_r_arcgis/index.md:22]:
```suggestion
This particular project has many components and is largely working in ArcGIS Pro's model builder with some deeper statistics and figure generation happening in RStudio. Our goal is to have cohesion between these two programs, so we did not want a workflow that required us to run some analysis in ArcGIS Pro, leave this program and open up RStudio to run a custom-made script, and then go back into ArcGIS Pro to run the rest of the model. Luckily, Esri's R-ArcGIS Bridge allows us to create the custom tool you'll see below and then import that into the ArcGIS Pro interface whenever we want to calculate the median download and upload speeds for each census block in our study area.  
```

File: content/blog/summarize_within_median_r_arcgis/index.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+```r
+require(sp) #classes and methods for spatial data, https://cran.r-project.org/web/packages/sp/index.html
+require(spatialEco) #needed for the point.in.polygon() function, https://cran.r-project.org/web/packages/spatialEco/index.html
+require(arcgisbinding) #initializes the R-ArcGIS bridge and connects your Esri license info
+
+#tie your license into the session
+arc.check_product()
+```
+Now we define the input and outputer parameters you want the user to provide via the tool interface in ArcGIS Pro:
```
Comment (merit-rluk) [inline content/blog/summarize_within_median_r_arcgis/index.md:48]:
```suggestion
Now we define the input and output parameters you want the user to provide via the tool interface in ArcGIS Pro:
```

File: content/blog/summarize_within_median_r_arcgis/index.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+```r
+  polygon_input <- in_params[(1)] #in this case, the census blocks
+  point_input <- in_params[(2)] #in this case, the points with the speed test data
+  attribute_input <- in_params[(3)] #this must be the EXACT name of the field in the data table
+  attribute_groupby <- in_params[(4)] #this must be the EXACT name of the field in the data table
+  coord_system <- in_params[(5)] #user chooses Point or Polygon from dropdown for the coordinate system they wnat to use
```
Comment (merit-rluk) [inline content/blog/summarize_within_median_r_arcgis/index.md:55]:
```suggestion
  coord_system <- in_params[(5)] #user chooses Point or Polygon from dropdown for the coordinate system they want to use
```

File: content/blog/summarize_within_median_r_arcgis/index.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+   #summarize points based on polygon ID via user input and update column names
+  summarized_median <- aggregate(x=pts.poly@data[attribute_input], by=list(pts.poly@data[ ,attribute_groupby]), FUN=median, drop=TRUE)
+  colnames(summarized_median)<-c(attribute_groupby,paste("Median",attribute_input))
+  
+  print("Median values aggregated")
+```
+We also will want the total number of points in each census block for a more robust understanding of our median metric:
+
+```r
+  #count total amount of points within each polygoj based on ID supplied by user
```
Comment (merit-rluk) [inline content/blog/summarize_within_median_r_arcgis/index.md:122]:
```suggestion
  #count total amount of points within each polygon based on ID supplied by user
```

File: content/blog/summarize_within_median_r_arcgis/index.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+  summarized_merged <- merge(summarized_median, summarized_count,by=attribute_groupby)
+  
+  #join table back to SpatialPolygonsDataFrame 
+  new_poly_spdf <- merge(poly.spdf, summarized_merged,by=attribute_groupby)
+  
+  #export to feature class
+  Output_FC <- arc.write(summarized_output, new_poly_spdf)
+```
+
+That's all the code! If you want to test this in RStudio before bringing it into ArcGIS Pro, simply comment out the initial function defintion and parameter defintions, define the parametrs as variables with the same name as the parameter defintion, and run the code from there.
```
Comment (merit-rluk) [inline content/blog/summarize_within_median_r_arcgis/index.md:141]:
```suggestion
That's all the code! If you want to test this in RStudio before bringing it into ArcGIS Pro, simply comment out the initial function definition and parameter definitions, define the parameters as variables with the same name as the parameter definition, and run the code from there. 
```

File: content/blog/summarize_within_median_r_arcgis/index.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+For more information on creating a custom tool from your script in ArcGIS Pro, visit this site: https://pro.arcgis.com/en/pro-app/latest/arcpy/geoprocessing_and_python/adding-a-script-tool.htm
+
+#Closing thoughts
+At the end of the day it is so important to stay true to your data, even if that means stepping outside of the preconfigured dropdowns your software provides. Some of the time your data may be normally distributed and you can use ArcGIS Pro's 'Summarize Within' with no issues. In the cases where your data is non-parametric, doing a bit of extra work to seek out a non-parametic solution will maintain the integrity of your data and the R-ArcGIS Bridge makes this so much more doable.
+
+It is also worth noting that there are other way to deal with non-normally distibuted data, such as removing outliers or transforing your data, that may work better for your specific project. The most important rule of thumb is to always use your brain first about what works best for the intended use of that data.
```
Comment (merit-rluk) [inline content/blog/summarize_within_median_r_arcgis/index.md:148]:
```suggestion
It is also worth noting that there are other way to deal with non-normally distributed data, such as removing outliers or transforming your data, that may work better for your specific project. The most important rule of thumb is to always use your brain first about what works best for the intended use of that data.
```

**Review by merit-dsclair — APPROVED** [review]

**Review by merit-ddennis — APPROVED** [review]

**Review by merit-schelcj — CHANGES_REQUESTED** [review]
Need to run through a linter, multiple issues found with the formatting.

**Comment by merit-mthompson** [comment]
@merit-amystuyv for markdown linting/checking you can use "markdownlint" plugin with VS Code.

Nearly all of the linting-related suggestions are whitespace around headings/paragraphs. The errors you can ignore are those for "single-h1" or "multiple level-one headings" as in the final blog post we want those level-one headings!

Also safe to ignore are ones about heading capitalization and other writing/style-related things.

**Comment by merit-amystuyv** [comment]
@merit-schelcj formatting issues have been fixed

**Review by merit-schelcj — APPROVED** [review]

**Pull request merged**